test/src/index/testChromeOptions.js

```diff
@@ -0,0 +1,114 @@
+const assert = require('assert');
+const Nightwatch = require('../../lib/nightwatch.js');
+const ChromeOptions =  require('selenium-webdriver/chrome').Options;
+
+describe('Test chrome options', function () {
+
+  it('Chrome option object with headless', function(){
+    const chromeOptions =  new ChromeOptions();
+    chromeOptions.headless();
+    const client = Nightwatch.createClient({
+      webdriver: {
+        options: chromeOptions
+      },
+      desiredCapabilities: {
+        browserName: 'chrome'
+      }
+    });
+    const options =  client.transport.createOptions();
+
+    assert.strictEqual(options instanceof ChromeOptions, true);
+    assert.deepStrictEqual(options.options_.args, ['headless']);
+  });
+
+  it('goog:chromeOptions detach driver option', function () {
+    const client = Nightwatch.createClient({
+      desiredCapabilities: {
```

Comment:
I think, this should be `Capabilities`, as `desiredCapabilities` is how we know it's a `jsonwire` protocol and it's not a `w3c` protocol

Reply:
It's not related to `w3c` or `jsonwire`. Nightwatch takes browser name from `desiredCapabilites` Object. You can checkout this for more details: https://github.com/nightwatchjs/nightwatch/blob/7e5edc6b32b00280565d3d2679df73e71b191582/lib/transport/transport.js#L314